### PR TITLE
Working demo for Login, JWT auth

### PR DESCRIPTION
**Pull request by ronald-so**
Not ready to merge yet! Implemented basic login functionality at `/login`, authorization using JWT tokens. Still working on the front-end side of things as well as the registration feature. Currently using a very basic JSON file as a database for user emails and passwords.

Known issues:
* Passwords should not be stored as plaintext
* `SECRET_KEY` should be randomly generated

I'm also unsure if I should be storing the JWT token using Flask's session handling (as I've done), or there is a better way to store the token into `localStorage` or `sessionStorage`

### CODE REVIEW

File: server/app.py [hunk 1]
```diff
@@ -1,12 +1,78 @@
...
+
+# Check if user is in our json "database" and return true if password matches
+# ***** Needs to be changed once actual db is set up *****
+def authenticate(email, password):
+    filtered_users = [x for x in users if (x['email'] == email)]
+
+    if not filtered_users: # no email match
+        return False
+    else:
+        if filtered_users[0]['password'] == password:
```
Comment (bonnieli) [inline server/app.py]: you can just rewrite this to one line: 

`return filtered_users[0]['password'] == password`

File: server/app.py [hunk 2]
```diff
@@ -1,12 +1,78 @@
...
+
+
+# Routes
+@app.route('/home')
+def home():
+    if 'token' in session:
+        decoded = decode(session['token'])
+        return render_template('home.html', logged_in=True, email=decoded)
+    else:
+        return render_template('home.html', logged_in=False)
```
Comment (bonnieli) [inline server/app.py]:
```
logged_in = False

if 'token' in session:
  decoded = decode(session['token'])
  logged_in = True

return render_template('home.html', logged_in=logged_in, email=decoded)
```

Comment (ronald-so) [reply server/app.py]: thanks! made those changes

**Comment by ronald-so** [comment]
Finished a basic login page for the front end, token now persists in localStorage and can be cleared using log out. Let me know if you guys find any issues or problems with it!

**Pull request merged**